File: Part 6 - Reinforcement Learning/Section 32 - Upper Confidence Bound (UCB)/random_selection.R
# Importing dataset
dataset = read.csv("Ads_CTR_Optimisation.csv")

# Implementing Random Selection
N = nrow(dataset)
d = ncol(dataset)
ads_selected = vector()
total_reward = 0
for (n in 1:N){
  ad = sample(d, 1)
  ads_selected = c(ads_selected, ad)
  reward = dataset[n,ad]
  total_reward = total_reward + reward
}

# Visualizing the results
hist(ads_selected,
     col = "blue",
     main = "Histogram of ads selections",
     xlab = "Ads",
     ylab = "Number of times each ad was selected")

File: Part 6 - Reinforcement Learning/Section 32 - Upper Confidence Bound (UCB)/random_selection.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
# entre [] se por o index das colunas que você quer, é um outro modo
dataset = pd.read_csv("Ads_CTR_Optimisation.csv")

# Implementing Random Selection
import random
N = dataset.shape[0]
d = dataset.shape[1]
ads_selected = []
total_reward = 0
for n in range(N):
    ad = random.randrange(d)
    ads_selected.append(ad)
    reward = dataset.values[n,ad]
    total_reward = total_reward + reward

# Visualizing the results
plt.hist(ads_selected)
plt.title("Histogram of ads selections")
plt.xlabel("Ads")
plt.ylabel("Number of times each ad was selected")
plt.show()